app/src/main/java/com/example/sharingang/utils/ImageAccess.kt

```diff
@@ -0,0 +1,188 @@
+import androidx.core.content.FileProvider
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.*
+
+
+class ImageAccess(
+    private val registry: FragmentActivity
+) :
+    DefaultLifecycleObserver {
+
+    private var storagePermissionGranted: Boolean = false
+    private var cameraPermissionGranted: Boolean = false
+    private lateinit var requestStoragePermissionLauncher: ActivityResultLauncher<String>
+    private lateinit var requestCameraPermissionLauncher: ActivityResultLauncher<String>
+
+    private lateinit var pickImage: ActivityResultLauncher<String>
+    private lateinit var takePicture: ActivityResultLauncher<Uri>
+
+    private var cameraUri: Uri? = null
+    private var imageUri: Uri? = null
+
+    private lateinit var imageView: ImageView
+
+    override fun onCreate(owner: LifecycleOwner) {
+        setupPermissionLaunchers(owner)
+        pickImage = registry.activityResultRegistry.register(
+            "openGallery",
+            owner,
+            ActivityResultContracts.GetContent(),
+            ::galleryCallback
+        )
+        takePicture = registry.activityResultRegistry.register(
+            "takePicture",
+            owner,
+            ActivityResultContracts.TakePicture(),
+            ::cameraCallback
+        )
+    }
+
+    fun unregister() {
+        pickImage.unregister()
+        requestStoragePermissionLauncher.unregister()
+        takePicture.unregister()
+        requestCameraPermissionLauncher.unregister()
+    }
+
+    fun setupImageView(view: ImageView) {
+        imageView = view
+    }
+
+    private fun setupPermissionLaunchers(owner: LifecycleOwner) {
+        requestStoragePermissionLauncher = createPermissionResultRegistry(
+            "storagePermission",
+            owner,
+            Manifest.permission.READ_EXTERNAL_STORAGE
+        )
+        requestCameraPermissionLauncher =
+            createPermissionResultRegistry("cameraPermission", owner, Manifest.permission.CAMERA)
+    }
+
+    private fun createPermissionResultRegistry(
+        key: String,
+        owner: LifecycleOwner,
+        permission: String
+    ): ActivityResultLauncher<String> {
+        return registry.activityResultRegistry.register(
+            key,
+            owner,
+            ActivityResultContracts.RequestPermission()
+        ) { isGranted: Boolean ->
+            if (!isGranted)
+                Toast.makeText(
+                    registry,
+                    "$permission permission successfully denied. Feature is disabled.",
+                    Toast.LENGTH_LONG
+                ).show()
+            when (permission) {
+                Manifest.permission.CAMERA -> cameraPermissionGranted = isGranted
+                Manifest.permission.READ_EXTERNAL_STORAGE -> storagePermissionGranted = isGranted
```

Comment:
Didn't we say that we should avoid the use of boolean values?

Reply:
The problem we had was not that we were using a boolean, it was that we were not checking the permissions everytime, but here it is done. Maybe I could do without it, but not sure.